src/migration/PoC_dfusion.js

```diff
@@ -10,49 +12,39 @@ async function migrate({ artifacts, deployer, network, account, web3, maxTokens
       account,
       web3,
     })
+    BiMap = artifacts.require("IdToAddressBiMap")
+    IterableAppendOnlySet = artifacts.require("IterableAppendOnlySet")
+    const TokenOWLProxy = artifacts.require("TokenOWLProxy")
+    feeToken = await TokenOWLProxy.deployed()
+    await deployer.deploy(BiMap)
+    await deployer.deploy(IterableAppendOnlySet)
+  } else {
+    BiMap = getArtifactFromNpmImport("@gnosis.pm/solidity-data-structures/build/contracts/IdToAddressBiMap", deployer, account)
+    IterableAppendOnlySet = getArtifactFromNpmImport(
+      "@gnosis.pm/solidity-data-structures/build/contracts/IterableAppendOnlySet",
+      deployer,
+      account
+    )
+    const TokenOWLProxy = getArtifactFromNpmImport("@gnosis.pm/owl-token/build/contracts/TokenOWLProxy", deployer, account)
+    feeToken = await TokenOWLProxy.deployed()
+    await BiMap.deployed()
+    await IterableAppendOnlySet.deployed()
```

Comment:
Could this logic be simplified to:

```js
BiMap = getArtifact(...)
IterableAppendOnlySet = getArtifact(...)
TokenOwl = getArtifact(...)
BatchExchange = getArtifact(...)

let fee_token // I'd slightly prefer if we also have instances for the other dependencies for linking but I guess it doesn't matter

// In development mode, we deploy new instances, otherwise use already deployed ones.
if (isDevelopmentNetwork(network) {
 feeToken = deployOwl(...)
 await deployer.deploy(BiMap)
 await deployer.deploy(IterableAppendOnlySet)
else {
 feeToken = await TokenOWLProxy.deployed()
 await  BiMap.deployed()
 await  IterableAppendOnlySet.deployed()
}
``

Reply:
we need the different artifacts, as explained above